### DISCUSSION TITLE
retest test_unixsock fails, bind(): Operation not permitted

### DISCUSSION THREAD

**Discussion by jimmbelll** [baresip/re#1021] [Q&A]
I see this failure with re-3.7.0 on debian bookworm.
`# ./build/test/retest -rv`
`# sudo ./build/test/retest -rv`

> test 170 -- test_unixsock
> unixsock: bind(): Operation not permitted [1] (http_njkjeqTj8Mzg1pPssEbxcWHcSR3WbNa.sock:0)
> 
> unixsock: TEST_ERR: .../re-3.7.0/test/unixsock.c:49: (Operation not permitted [1])
> test: test_unixsock: test failed (Operation not permitted [1])
> 
> retest: TEST_ERR: .../re-3.7.0/test/main.c:238: (Operation not permitted [1])

**Comment by sreimers** [comment]
This looks like a real permission problem. Linux kernel has this [CAP_NET_ADMIN](https://man7.org/linux/man-pages/man7/capabilities.7.html) unix socket check:

https://elixir.bootlin.com/linux/latest/source/net/unix/af_unix.c#L2990

**Reply by jimmbelll** [reply]
**This fix works:**

re/test/unixsock.c, row 43:
`	re_snprintf(socket, sizeof(socket), "unix:/tmp/http_%s.sock", filename);`

It could be that I'm building from within a cifs mount, so creating a unix socket there is verboten.
